src/components/SettingsPage.vue

```diff
@@ -243,18 +243,82 @@
         </div>
       </div>
       <!--End of Accordion-->
-      <button class="btn btn-primary text-success ml-3 mt-2" type="button"><a style='color:green'>Save</a></button>
-            <button class="btn btn-primary text-success ml-3 mt-2" type="button"><a style='color:red'>Discard</a></button>
-
+      <button
+        class="btn btn-primary text-success ml-3 mt-2"
+        type="button"
+        v-on:click="handleSubmit"
+      >
+        <a style="color:green">Save</a>
+      </button>
+      <button class="btn btn-primary text-success ml-3 mt-2" type="button">
+        <a style="color:red">Discard</a>
+      </button>
     </div>
   </div>
 </template>
 
 <script>
 import AdvancedSettings from "@/components/AdvancedSettings.vue";
+import { config } from "@/configurations";
+
 export default {
   name: "SettingsPage",
   components: { AdvancedSettings },
+  data() {
+    return {
+      profilePicture: "",
+      username: "",
+      bio: "",
+      password1: "",
+      password2: "",
+      submitted: false,
+    };
+  },
+  methods: {
+    handleSubmit() {
+      this.submitted = true;
+      const { profilePicture, username, bio, password1, password2 } = this;
+      const { dispatch } = this.$store;
+      let updateFields = {};
+      if (bio) {
+        updateFields["bio"] = bio;
+      }
+      if (profilePicture) {
+        updateFields["profilePicture"] = profilePicture;
+      }
+      if (username) {
+        updateFields["username"] = username;
+      }
+      if (password1) {
+        updateFields["password1"] = password1;
+      }
+      if (password2) {
+        updateFields["password2"] = password2;
+      }
+
+      if (password1 && password1.trim() !== password2.trim()) {
+        dispatch(
+          "alertStore/error",
+          config.messagingConfig.messages.error.password_mismatch,
+          { root: true }
+        );
+      }
+      if (bio && bio.length >= 200) {
+        dispatch(
+          "alertStore/error",
+          config.messagingConfig.messages.error.bio_size,
+          { root: true }
+        );
+      }
+      dispatch("userStore/setting", {
```

Comment:
please make use of `updateFields` instead of passing an entire list of variables which are non-mandatory.